string timestamp
int64[] err